go-controller/pkg/ovn/egressip.go

```diff
@@ -1221,14 +1268,30 @@ func (oc *Controller) syncStaleSNATRules(egressIPToPodIPCache map[string]sets.St
 // atomic items with the same general information repeated across most (egressIP
 // name, logical IP defined for that name), hence use a cache to avoid round
 // trips to the API server per item.
-func (oc *Controller) generatePodIPCacheForEgressIP(eIPs []interface{}) (map[string]sets.String, error) {
+func (oc *Controller) generatePodIPCacheForEgressIP(eIPs []interface{}) (map[string]sets.String, map[string]sets.String, map[string]sets.String, error) {
 	egressIPToPodIPCache := make(map[string]sets.String)
+	egressIPToPodIPCacheNextHops := make(map[string]sets.String)
```

Comment:
i'm getting a bit confused by all these caches and the names don't really make sense to me. I'm thinking maybe it would be easier to make a single cache:
```
type egressIPCacheEntry struct {
   nodeName string
   egressIP    net.ip
   name         string
}

// egressIPCache holds a key of podIP to cache entry
egressIPCache map[net.ip]egressIPCacheEntry

```
wdyt?

Reply:
Yes, good idea @trozet . Let me work on that and refresh this PR.

Reply:
This may be easier to follow:

```
type egressIPCacheEntry struct {
       podIPs           sets.String
       gatewayRouterIPs sets.String
       egressIPs        sets.String
}
```